### PR TITLE
Org Supplier Relationship

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] Branched From - https://github.com/andygray/adminfire/pull/16
* [x] CRUD Sites Service
* [x] Test `SyncArray` with `FK_SiteID` within `organisation` collections :-1: 
* [x] Test `SyncArray` with `FK_OrganisationID` within `sites` collections :+1: 
* [x] Add `Site` entity `AuditLog` entries
* [x] Add `organisation` site reference actions to `AuditLogs` 
* [x] Add notifications for site CRUD operations
* [x] Refactor duplicate `AuditLog` code
* [x] Ready for review

### CODE REVIEW

File: app/js/entity/sites/SitesService.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            var audit = function () {
+                $log.debug('Updated site');
+            };
+
+            var logError = function (error) {
+                $log.error('Failed to update site', error);
+            };
+
+            // If loaded 3 way bound object simply call create
+            if (_.isFunction(site.$save)) {
```
Comment (andygray) [inline app/js/entity/sites/SitesService.js]: I imagine you googled this?

File: app/js/ui-routes.js [hunk 2]
```diff
@@ -95,13 +100,24 @@
...
-                                return new Organisation(organssation);
+                            .then(function (organsation) {
+                                return new Organisation(organsation);
                             });
                     },
                     /**
                      * @ngInject
                      */
+                    sites: function ($log, $q, SitesService, organisation) {
+                        var sites = organisation.sites().map(function (siteId) {
```
Comment (andygray) [inline app/js/ui-routes.js]: I like it. Bit like hibernate - though if you don't need the sites on a "page" then you don't need to resolve ID array

**Comment by andygray** [comment]
:+1: Looks good.

File: app/js/entity/sites/SitesService.js (hunk 1 above)
Comment (jamesmorgan) [inline app/js/entity/sites/SitesService.js]: I'm not sure its needed. All objects should be `$FirebaseObject` I think its a hang over from an old implementation so could maybe go. It does work as described and you can save existing objects if the correct `$id` ever if they are not  `$FirebaseObject`.

**Comment by jamesmorgan** [comment]
@andygray I have fixed the bug we spoke about and its ready to be merged at/reviewed.

File: app/js/entity/organisation/Organisation.js [hunk 3]
```diff
@@ -61,6 +61,47 @@
...
                 return this;
+            },
+
+            /**
+             * Adds the provided site ID to this {@link Organisation}
+             *
+             * @param siteId
+             * @return {Organisation} itself
+             */
+            withSiteReference: function (siteId) {
```
Comment (andygray) [inline app/js/entity/organisation/Organisation.js]: nice and simples..

**Comment by andygray** [comment]
:+1: World class!

**Pull request merged**